src/test/java/org/sonar/plugins/stash/StashIssueReportingPostJobTest.java

```diff
@@ -247,6 +252,36 @@ public void testExecuteOnWithNoDiffReport() throws Exception {
     verify(stashRequestFacade, times(0)).resetPullRequestApproval(eq(pr), (StashClient)Mockito.anyObject());
   }
 
+  @Test
+  public void testShouldApprovePullRequest() {
+    Issue minorIssue = new DefaultIssue().setSeverity(Severity.MINOR);
+    Issue majorIssue = new DefaultIssue().setSeverity(Severity.MAJOR);
+
+    report = new ArrayList<>();
+
+    report.add(minorIssue);
+    report.add(majorIssue);
+
+    assertFalse(
+        StashIssueReportingPostJob.shouldApprovePullRequest(Optional.empty(), report)
+    );
+
+    report.clear();

```

Comment:
The test with multiple asserts might fail at first assertion and other assertions are not executed at all.

It is good when one break makes only one assert fail, letting other asserts be executed in separate test methods.

As soon as you break test code to separate tests, you have a problem of repetitous setup. It is solved easily by single `@Before` method or a helper method called in each method.
If the assert should be checked only after some other checks have succeeded (which are checked in separate test methods) then one can use `assume*` methods of JUnit.

As soon as you have your tests split this way, single defect breaks one test only. Some other tests may become yellow telling you the defect is essential for checks performed there. The good thing you'd know what have failed exactly. If several things have failed you'd know all of them.